app/models/profile.rb

```diff
@@ -14,4 +14,30 @@ class Profile < ActiveRecord::Base
   validates :user, presence: true
   validates_presence_of :first_name, :last_name, :gender, :birth_date, :email, :school, :street_name, :zip_code, :city, :state, :country
 
-end
\ No newline at end of file
+  # Returns the age of the user based on the current date
+  #
+  # @param none
+  # @return [Int] for age as number of years
+  def age
+    now = Time.now.utc.to_date
+    now.year - birth_date.year - ((now.month > birth_date.month || (now.month == birth_date.month && now.day >= birth_date.day)) ? 0 : 1)
```

Comment:
Thanks, now the bug should be fixed (+1 for adding test case). Maybe this algorithm could be separated by using descriptive intermediate variables. At least for me it needed some time to understand it. But I think it is "okish".